src/agent/src/main.c

```diff
@@ -724,7 +726,12 @@ _Bool StartupAgent(const ADUC_LaunchArguments* launchArgs)
 #else
         if (!GetConnectionInfoFromIdentityService(&info))
         {
-            goto done;
+            Log_Info("Failed to get connection information from AIS. Defaulting to configuration file");
+
+            if (!GetConnectionInfoFromADUConfigFile(&info))
```

Comment:
>GetConnectionInfoFromADUConfigFile [](start = 17, length = 34)

we will now fallback to config? perhaps some opportunity here for some minor refactor later

Reply:
That's the plan Nox laid out. 

I think we need a clearer story for our provisioning routes. It's becoming more like a web then a straight line.

---
In reply to: [594749103](https://github.com/Azure/iot-hub-device-update/pull/16#discussion_r594749103) [](ancestors = 594749103)

Reply:
what's the rationale? historically, I've been adverse to having any fallback as it tends to either confuse users, or introduce undesirable behaviors (what if there's a stale config file, and now you're connected to an endpoint you dont want to be connected to?)

---
In reply to: [594754375](https://github.com/Azure/iot-hub-device-update/pull/16#discussion_r594754375) [](ancestors = 594754375,594749103)